### PR TITLE
nfd-master: drop the -nrt-namespace cmdline flag

### PR DESCRIPTION
**Pull request by marquiz**
Always create the CR in the namespace we're running in. NFD owns the CR
so it should create it in its own namespace rather than messing around
in other namespaces. Also makes cleanup less error-prone. Deleting the
nfd namespace removes the CRs too, not leaving them abandoned in some
random corner of the cluster.

Update RBAC rules accordingly as nfd-master doesn't require cluster-wide
privileges to NodeResourceTopology objects, anymore.

### CODE REVIEW

File: deployment/base/rbac/master-clusterrole.yaml [hunk 1]
```diff
@@ -12,11 +12,3 @@ rules:
   - patch
   - update
   - list
-- apiGroups:
```
Comment (ArangoGutierrez) [inline deployment/base/rbac/master-clusterrole.yaml:15]: This `topology.node.k8s.io` changes don't align with the PR description

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
Looks good, just a minor question

File: deployment/base/rbac/master-clusterrole.yaml (hunk 1 above)
Comment (marquiz) [reply deployment/base/rbac/master-clusterrole.yaml:15]: Updated commit message

**Comment by marquiz** [comment]
Superseded/fixed by #652 
/close

**Pull request closed without merging**